dashboard/public/pixelation/pixelation.js

```diff
@@ -385,3 +385,25 @@ function showPNG() {
   w.focus();
   options.saveProject && options.saveProject();
 }
+
+function onFinishedButtonClick() {
+  var finishedButton = $('#finished');
+  if (finishedButton.attr('disabled')) {
+    return;
+  }
+  finishedButton.attr('disabled', true);
+
+  if (options.saveProject) {
+    options.saveProject(onSaveProjectComplete);
```

Comment:
In general it seems like using simple callbacks is ok when there is just 1 (or maybe 2) of them, but once there are more than 2 or 3 in a chain (or tree), that moving to Promises is a good escalation path. In this case I agree that it would be good to move project.save (and processResults, or whatever it ends up as) to return a Promise soon.

I'm open to trying the ES6 stuff but that seems like a bigger decision :)